controller/lifecycle/lifecycle.go

```diff
@@ -371,3 +393,11 @@ func (l *LifecycleManager) SetupWithManager(mgr ctrl.Manager, maxReconciles int,
 		WithEventFilter(predicate.And(eventPredicates...)).
 		Complete(r)
 }
+
+// WithPrepareContextFunc allows to set a function that prepares the context before each reconciliation
+// This can be used to add additional information to the context that is needed by the subroutines
+// You need to return a new context and an OperatorError in case of an error
+func (l *LifecycleManager) WithPrepareContextFunc(prepareFunction func(ctx context.Context, instance RuntimeObject) (context.Context, errors.OperatorError)) *LifecycleManager {
```

Comment:
🟢 One small suggestion would be instead of repeating the type in the struct and in this function to extract it into a `PrepareContextFunc` type. But thats just a suggestion and pretty nitpicky 😄

Reply:
i think its a good suggestion 👍 
Pushed an update with it